lab5/utils/CollectionManager.java

```diff
@@ -0,0 +1,390 @@
+                int month = scanner.nextInt();
+                if (month < 1 || month > 12) {
+                    throw new Exception("write month only within 1 to 12");
+                }
+
+                //input for day
+                System.out.println("Enter the day: ");
+                int day = scanner.nextInt();
+                if(month == 1 || month == 3 || month == 5 || month == 7 || month == 8 || month == 10 || month == 12) {
+                    if (day > 31) throw new Exception("Out of range date in the month");
+                } else if(month == 4 || month == 6 || month == 9 || month == 11) {
+                    if (day > 30) throw new Exception("Out of range date in the month");
+                } else if(month == 2){
+                    if (year % 4 == 0) {
+                        if (day > 29) throw new Exception("Out of range date in the month");
+                    } else {
+                        if (day > 28) throw new Exception("Out of range date in the month");
+                    }
+                }
+
+                birthday = LocalDateTime.of(year, month, day, 0, 0);
+                validInput = true;
+            } catch (Exception e) {
+                System.out.println(e.getMessage());
+                scanner.nextLine();
+            }
+        }
+
+        return new Human(birthday);
+
+    }
+
+    public void clear() {
+        if(!cities.isEmpty()){
+            ArrayDeque<City> toRemove = new ArrayDeque<>(cities);
+            cities.removeAll(toRemove);
+            System.out.println("Collection has been cleared");
+        } else {
+            System.out.println("Collection is empty");
+        }
+
+    }
+
+    public void updateCity(City city) {
+        Scanner scanner = new Scanner(System.in);
+        System.out.print("Enter name: ");
+        String name = scanner.next();
+
+        Coordinates coordinates = addCoordinate(scanner);
+
+        float area = addArea(scanner);
+        Long population = addPopulation(scanner);
+        double metersAboveSeaLevel = addMetersAboveSeaLevel(scanner);
+        float agglomeration = addAgglomeration(scanner);
+        Climate climate = addClimate(scanner);
+        Government government = addGovernment(scanner);
+        Human governor = addGovernor(scanner);
+
+        city.setName(name);
+        city.setCoordinates(coordinates);
+        city.setArea(area);
+        city.setPopulation(population);
+        city.setMetersAboveSeaLevel(metersAboveSeaLevel);
+        city.setAgglomeration(agglomeration);
+        city.setClimate(climate);
+        city.setGovernment(government);
+        city.setGovernor(governor);
+
+        System.out.println("City with id: " + city.getId() + " has been updated");
+    }
+
+    public void removeCity(City city) {
+        cities.remove(city);
+        System.out.println("City with ID " + city.getId() + " has been removed");
+    }
+
+    public int countByAgglomeration(String arg) {
```

Comment:
Why do you pass a string here? 
The argument is float, the collection manager should know nothing about parsing